content/3_Silver/Binary_Search.mdx

```diff
@@ -1309,16 +1338,61 @@ int find_max_true(int l, int r, function<bool(int)> f) {
 	return l;
 }
 ```
+</CPPSection>
+<JavaSection>
+```java
+public int findMaxTrue(int l, int r, Predicate<Integer> f) {
+	while (l <= r) {  // <= instead of <
+		int m = (l + r) / 2;
+		if (f.test(m)) {
+			l = m;
+		} else {
+			r = m - 1;
+		}
+	}
+	return l;
+}
+```
+</JavaSection>
+<PySection>
+```py
+def find_max_true(l: int, r: int, f) -> int:
```

Comment:
done
note that `typing.Callable` is deprecated since Python 3.9
https://docs.python.org/3/library/typing.html#typing.Callable
See also: https://peps.python.org/pep-0585/ , though `collections.abc.Callable` also seems to bug in Python 3.9.1/2.

Reply:
sure, just 1-up me on my own comment why don’t you \s